File: app/src/main/java/com/example/geosolve/dataModel/Figure.java
package com.example.geosolve.dataModel;

import java.util.ArrayList;

public class Figure {
    private ArrayList<Line> lines;
    private ArrayList<Node> nodes;


    public Figure() {
        this.lines = new ArrayList<>();
        this.nodes = new ArrayList<>();
    }

    public void addNode(Node node) {
        nodes.add(node);
    }

    public ArrayList<Node> getNodes() {
        return nodes;
    }

    public boolean stopAllNode() {
        boolean answer = false;
        for (Node node : nodes) {
            answer = answer | node.stopMove();
        }
        return answer;
    }
}

File: app/src/main/java/com/example/geosolve/model/Figure.kt
package com.example.geosolve.model

import java.util.*

class Figure {

    private val lines: ArrayList<Line> = ArrayList()
    val nodes: ArrayList<Node> = ArrayList()

    fun stopAllNode(): Boolean {
        var answer = false
        for (node in nodes) {
            answer = answer or node.stopMove()
        }
        return answer
    }

    fun addNode(node: Node) {
        nodes.add(node)
    }

    fun delNode(touchX: Float, touchY: Float) {
        for (node in nodes)
            if (node.inRadius(touchX, touchY)) {
                val index: Int = nodes.indexOf(node)
                nodes.removeAt(index)
                break
            }
    }

    fun clearFigure(){
        nodes.clear()
        lines.clear()
    }
}